src/test/integration/models/subject.spec.js

```diff
@@ -0,0 +1,71 @@
+const { setupTestServer, stopServer } = require('~/test/setupSafeTest')
+const Subject = require('~/models/subject')
+const Category = require('~/models/category')
+
+describe('Subject model', () => {
+  let server
+
+  beforeAll(async () => {
+    const setup = await setupTestServer()
+    server = setup.server
+  })
+
+  afterAll(async () => {
+    await stopServer(server)
+  })
+
+  it('should include all required fields', async () => {
+    const subjects = await Subject.find({})
+
+    for (const subject of subjects) {
+      expect(subject).toHaveProperty('name')
+      expect(subject).toHaveProperty('category')
+      expect(subject).toHaveProperty('totalOffers')
+      expect(subject.totalOffers).toHaveProperty('student')
+      expect(subject.totalOffers).toHaveProperty('tutor')
+      expect(subject).toHaveProperty('createdAt')
+      expect(subject).toHaveProperty('updatedAt')
+    }
+  })
+
+  it('should have valid fields data types', async () => {
+    const subjects = await Subject.find({})
+
+    for (const subject of subjects) {
+      expect(typeof subject.name).toBe('string')
+      expect(typeof subject.category).toBe('object')
+      expect(typeof subject.totalOffers).toBe('object')
+      expect(typeof subject.totalOffers.student).toBe('number')
+      expect(typeof subject.totalOffers.tutor).toBe('number')
+      expect(subject.createdAt).toBeInstanceOf(Date)
+      expect(subject.updatedAt).toBeInstanceOf(Date)
+    }
+  })
+
+  it('should have unique name', async () => {
+    const subjects = await Subject.find({}).select({ name: 1, _id: 0 })
```

Comment:
Is it necessary to have _id: 0 here?
```suggestion
    const subjects = await Subject.find({}).select({ name: 1 })
```

Reply:
No. Just removed it as I only need names for that test. Do you think it would better to leave it (included in the result)?